api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -109,6 +109,8 @@ public Schema(int schemaId, List<NestedField> columns, Map<String, Integer> alia
   static void validateIdentifierField(int fieldId, Map<Integer, Types.NestedField> idToField,
                                               Map<Integer, Integer> idToParent) {
     Types.NestedField field = idToField.get(fieldId);
+    Preconditions.checkArgument(field != null,
+        "Can not add filedId %d as an identifier field: field not exists", fieldId);
```

Comment:
Nit: `Can not` -> `Cannot`.

RE the phrasing: In this case, the phrasing does match the rest of the file, though it's not what we normally use in Iceberg. Normally we use `{Problem} {Cause}` and possibly `{Problem} {Cause}: {relevant value}`. And the messages ideally read as plain English phrases.

So it's abnormal to see the `:` in the middle, but it does match the rest of the file. I would say `Cannot add identifier partition field from missing fieldId: %d` and then let the rest be inferred from the stack trace, as adding a partition field is what the user is trying to do here.

But in this case, your phrasing is similar to the rest of the file so it's not the worst phrasing.

So I'd update `Cannot` and `fieldId` typos and then add the test. The phrase is easy to update, but the test is needed regardless.